### PR TITLE
changed the way we report live analysis to task center

### PR DESCRIPTION
**Pull request by heejaechang**
previously, we listen to the diagnostic service to report progress. problem is that it only raises events if it found errors on a file. so what we report is actually the last file we found errors on rather than the file that we are analyzing.

this caused confusion since we report in task center that we are analyzing file "A" when it is actually "analyzed" not "analyzing"

another issue is since it only reports file that contains errors. we might not actually show anything in task center if there is no error, or show file "A" for a long time if that is only filed with errors.

this PR changes the experience closer to what users would expect. and now progress is for solution crawler not specifically on diagnostics.

now we report file that solution crawler is analyzing.

there is still caveat such as solution crawler can pause between processing a file if VS is busy. but it will still show file "A". or we will not update UI at least 200ms apart and etc.

since it is task center where we don't want to be too impactful to VS, based on feedback we will see whether we need to do more such as detect solution crawler pause and update task center to show a pause. or update task center to show different stage such as analyzing/analyzed.

or show in task center, what analyzer is actually running such as diagnostic, todo, designer attribute scan, find all reference cache and etc.

old
progress2

new
progress

### CODE REVIEW

File: src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs [hunk 1]
```diff
@@ -18,11 +19,28 @@ internal interface ISolutionCrawlerProgressReporter
...
         /// Notifications for this event are serialized to preserve order. 
         /// However, individual event notifications may occur on any thread.
         /// </summary>
-        event EventHandler<bool> ProgressChanged;
+        event EventHandler<ProgressData> ProgressChanged;
+    }
+
+    internal struct ProgressData
+    {
+        public Status Type { get; }
```
Comment (mavasani) [inline src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs]: Nit: Probably better to call the property also `Status`

File: src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs [hunk 2]
```diff
@@ -18,11 +19,28 @@ internal interface ISolutionCrawlerProgressReporter
...
+        public string FilePath { get; }
+
+        public ProgressData(Status type, string filePath)
```
Comment (mavasani) [inline src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs]: Looking at the call sites below it seems the `filePath` can be null. Can you add comments and/or call it `FilePathOpt`, and add comments when null is expected?

Comment (heejaechang) [reply src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs]: okay.

File: src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs (hunk 1 above)
Comment (heejaechang) [reply src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs]: I get compiler error saying Status is a type.

**Pull request merged**